lottie-swift/src/Private/LayerContainers/Utility/LayerImageProvider.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  LayerImageProvider.swift
+//  lottie-swift
+//
+//  Created by Brandon Withrow on 1/25/19.
+//
+
+import Foundation
+
+/// Connects a LottieImageProvider to a group of image layers
+class LayerImageProvider {
+  
+  var imageProvider: AnimationImageProvider {
+    didSet {
+      reloadImages()
+    }
+  }
+  
+  fileprivate(set) var imageLayers: [ImageCompositionLayer]
+  let imageAssets: [String : ImageAsset]
+  
+  init(imageProvider: AnimationImageProvider, assets: [String : ImageAsset]?) {
+    self.imageProvider = imageProvider
+    self.imageLayers = [ImageCompositionLayer]()
+    if let assets = assets {
+      self.imageAssets = assets
+    } else {
+      self.imageAssets = [:]
+    }
+    reloadImages()
+  }
+  
+  func addImageLayers(_ layers: [ImageCompositionLayer]) {
+    for layer in layers {
```

Comment:
Not sure if it's too unreadable but this can be written with a two-liner:

let foundLayers = layers.filter { imageAssets[$0.imageReferenceID] != nil }
imageLayers.append(contentsOf: foundLayers)